test/ContractInteractor.test.ts

```diff
@@ -0,0 +1,142 @@
+        let _createForwarderStub: sinon.SinonStub;
+        const fakeIForwarderInstance: sinon.SinonStubbedInstance<IForwarderInstance> &
+            IForwarderInstance = stubInterface<IForwarderInstance>();
+        const fakeSuffixData = 'fakeSuffix';
+        const fakeRelayRequest: RelayRequest = {
+            request: {
+                to: 'fake_address',
+                data: 'fake_data',
+                gas: '1'
+            } as ForwardRequest,
+            relayData: {
+                gasPrice: '0',
+                callForwarder: 'fake_address'
+            } as RelayData
+        };
+        const fakeSignature = 'fake_signature';
+
+        before(() => {
+            _createForwarderStub = sinon
+                .stub(contractInteractor, '_createForwarder')
+                .callsFake(() => Promise.resolve(fakeIForwarderInstance));
+        });
+
+        it('should fail if suffixData is null', async () => {
+            const error = new TypeError(
+                "Cannot read properties of null (reading 'substring')"
+            );
+            fakeIForwarderInstance.verify.throwsException(error);
+            await assert.isRejected(
+                contractInteractor.verifyForwarder(
+                    null,
+                    fakeRelayRequest,
+                    fakeSignature
+                ),
+                error.message
+            );
+        });
+
+        it('should fail if RelayRequest is null', async () => {
+            const error = new TypeError(
+                "Cannot read properties of null (reading 'relayData')"
+            );
+            fakeIForwarderInstance.verify.throwsException(error);
+            await assert.isRejected(
+                contractInteractor.verifyForwarder(
+                    fakeSuffixData,
+                    null,
+                    fakeSignature
+                ),
+                error.message
+            );
+        });
+
+        it('should fail if Signature is null', async () => {
+            const error = new TypeError(
+                "Cannot read properties of null (reading 'length')"
+            );
+            fakeIForwarderInstance.verify.throwsException(error);
+            await assert.isRejected(
+                contractInteractor.verifyForwarder(
+                    fakeSuffixData,
+                    fakeRelayRequest,
+                    null
+                ),
+                error.message
+            );
+        });
+
+        it('should fail if callForwarder is null', async () => {
+            _createForwarderStub.restore();
+            fakeRelayRequest.relayData.callForwarder = null;
+            await assert.isRejected(
+                contractInteractor.verifyForwarder(
+                    fakeSuffixData,
+                    fakeRelayRequest,
+                    fakeSignature
+                ),
+                'Invalid address passed to IForwarder.at(): null'
+            );
+        });
+    });
+});
```

Comment:
I suggest to add testing for the happy path and the possible errors the contract can emit.